application/client.core/src/app/environment/components/sidebar/shell/environment/environment_variables/component.ts

```diff
@@ -0,0 +1,198 @@
+
+    private _prevValue: string = '';
+    private _prevVariable: string = '';
+    private _editing: IEnvironment | undefined;
+
+    constructor() { }
+
+    public ngOnInit() {
+        this._sortInformation();
+        this._ng_height = Math.round(window.innerHeight * .7);
+    }
+
+    public ngOnDestroy() {
+        if (this._ng_selected !== undefined) {
+            this._ng_selected.editing = {
+                value: false,
+                variable: false
+            };
+            this._ng_selected = undefined;
+        }
+        this._ng_adding = false;
+        this.setEnvironment(this.information);
+    }
+
+    public _ng_onContexMenu(event: MouseEvent, env: IEnvironment, type: string) {
+        const items: IMenuItem[] = [
+            {
+                caption: 'Edit',
+                handler: () => {
+                    this._changeFocus(env);
+                    if (type === this._ng_value) {
+                        env.editing.value = true;
+                        this._prevValue = env.value;
+                    } else {
+                        env.editing.variable = true;
+                        this._prevVariable = env.variable;
+                    }
+                }
+            },
+            {
+                caption: 'Remove',
+                handler: () => {
+                    this._changeFocus();
+                    this.information.env = this.information.env.filter((ngEnv: IEnvironment) => {
+                        return ngEnv.variable !== env.variable && ngEnv.value !== env.value;
+                    });
+                },
+            },
+        ];
+        ContextMenuService.show({
+            items: items,
+            x: event.pageX,
+            y: event.pageY,
+        });
+    }
+
+    public _ng_onKeyUp(event: KeyboardEvent) {
+        if (event.key !== 'Enter') {
+            return;
+        }
+        this._finishCreating(true);
+        this._setCustomStatus();
+        this._sortInformation();
+    }
+
+    public _ng_onClick(env: IEnvironment) {
+        if (this._ng_selected !== env) {
+            this._changeFocus();
+        }
+        if (this._ng_selected !== undefined) {
+            this._ng_selected.selected = false;
+        }
+        this._ng_selected = env;
+        this._ng_selected.selected = true;
+    }
+
+    public _ng_onDoubleClick(env: IEnvironment, type: string) {
+        this._changeFocus(env);
+        if (type === this._ng_value) {
+            env.editing.value = true;
+            this._prevValue = env.value;
+        } else {
+            env.editing.variable = true;
+            this._prevVariable = env.variable;
+        }
+    }
+
+    public _ng_add() {
+        this._ng_selected = undefined;
+        this._ng_adding = true;
+        if (this._ng_variableAdd !== undefined) {
+            this._ng_variableAdd.nativeElement.scrollIntoView();
+        }
+    }
+
+    public _ng_remove() {
+        if (this._ng_selected !== undefined) {
```

Comment:
btw `_ng_selected` cannot be `undefined` according to component's declarations
```
public _ng_selected: IEnvironment;
```
probably checks are stay from previous edition of component